glide-core/redis-rs/redis/src/aio/mod.rs

```diff
@@ -181,6 +220,16 @@ where
         }
     }
 
+    if discover_az {
+        if let Err(e) = update_az_from_info(con).await {
+            return Err(RedisError::from((
+                ErrorKind::ResponseError,
+                "Failed to update AZ from info",
+                format!("{:?}", e),
+            )));
+        }
```

Comment:
```suggestion
if discover_az {
    update_az_from_info(con).await?;
}
```

You are already building the same error inside `update_az_from_info`, just propagate it